open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3818,176 @@ components:
+
+    IncrementalSnapshotScanContext:
+      description:
+        Context for scanning data appended in a range of snapshots.
+        The scan always follows the schema of the snapshot at the main branch head.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/BaseTableScanContext'
+      required:
+        - type
+        - start-snapshot-id
+      properties:
+        type:
+          type: string
+          enum: ["incremental-snapshot-scan"]
+        select:
+          $ref: '#/components/schemas/SelectedFieldNames'
+        filter:
+          $ref: '#/components/schemas/Filter'
+        case-sensitive:
+          description: If field selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        start-snapshot-id:
+          description: The ID of the starting snapshot of the incremental scan
+          type: integer
+          format: int64
+        inclusive-start:
+          description: If the data appended in the start snapshot should be included in the scan
+          type: boolean
+          default: false
+        end-snapshot-id:
+          description:
+            The ID of the inclusive ending snapshot of the incremental scan.
+            If not specified, the snapshot at the main branch head will be used as the end snapshot.
+          type: integer
+          format: int64
+
+    FieldName:
+      description:
+        A field name that follows the Iceberg naming standard, and can be used in APIs like
+        Java `Schema#findField(String name)`.
+
+        The nested field name follows these rules
+        - nested struct fields are named by concatenating field names at each struct level using dot (`.`) delimiter,
+        e.g. employer.contact_info.address.zip_code
+        - nested fields in a map key are named using the keyword `key`, e.g. employee_address_map.key.first_name
+        - nested fields in a map value are named using the keyword `value`, e.g. employee_address_map.value.zip_code
+        - nested fields in a list are named using the keyword `element`, e.g. employees.element.first_name
+      type: string
+
+    SelectedFieldNames:
+      description:
+        A list of fields in schema that are selected in a table scan.
+        When not specified, all columns in the requested schema should be selected.
+      type: array
+      items:
+        $ref: '#/components/schemas/FieldName'
+
+    Filter:
+      description:
+        an unbounded expression to describe the filters to apply to a table scan, 
+        default to `TrueExpression` meaning that nothing is filtered.
+      allOf:
+        - $ref: '#/components/schemas/Expression'
+      default: { "type": "true" }
+
+    PlanTask:
+      description:
+        An opaque JSON object that contains information provided by the REST server
+        to be utilized by clients for distributed table scan planning; should be supplied
+        as input in `PlanTable` operation.
+      type: object
```

Comment:
@Fokko yes. I think `binary` is more appropriate. The intent here is that `PlanTask` should remain opaque to the client. The client receives the Preplan response with this `PlanTask` object, and call `plan` API with the same `PlanTask` object.

Reply:
Although since this is being serialized/deserialized to/from JSON as part of `PreplanTableResponse` and `PlanTableRequest`, we probably want to encode it as Base64.

Reply:
Ah I see, thanks for the context, I'm jumping in here late. So there is no custom context, just a payload that would be passed to the `plan` method, that makes sense.

Yes, it could be an object or a `base85` encoded json blob ([more efficient than `base64`](https://docs.rs/base85/latest/base85/#:~:text=A%20base85%2Dencoded%20string%20is,and%20needing%20no%20character%20escapes.))

Reply:
Got it. I think we can change it to:
```
type: string
format: byte
```
, according to https://swagger.io/docs/specification/data-models/data-types/#string
> byte – base64-encoded characters, for example, U3dhZ2dlciByb2Nrcw==

We can update the documentation to say that this is supposed to be an opaque object in string, encoded in base64 or base85. Technically the service can decided whether to use base64 or base85.

I don't think it can be an `object`, since we don't actually define the concrete properties inside the object, so we cannot serialize/deserialize on the client side.

Reply:
I think that this is very likely to be JSON so it doesn't make much sense to me to allow a binary blob here. A string might make some sense to avoid needing to parse the JSON, but my main objection is that it requires the service to serialize twice. The first time produces a string that is then parsed a second time.

Using the original `object` type seems better to me, although I do see the issue for clients. On the client side, I'd probably just map this to generic `Map`, `List`, `String`, etc. or keep the Jackson representation (`JsonNode`) in an opaque object that is understood by our parser but doesn't expose the payload. This solution seems easy enough that I would go with `object` as my main preference, but if there's consensus around `string` I'm fine with that too.

Reply:
Got it. Thanks! I will give `JsonNode` a try.